stytch/consumer/api/oauth.py

```diff
@@ -0,0 +1,185 @@
+# !!!
+# WARNING: This file is autogenerated
+# Only modify code within MANUAL() sections
+# or your changes may be overwritten later!
+# !!!
+
+from __future__ import annotations
+
+from typing import Any, Dict, Optional
+
+from stytch.consumer.models.oauth import AttachResponse, AuthenticateResponse
+from stytch.core.api_base import ApiBase
+from stytch.core.http.client import AsyncClient, SyncClient
+
+
+class OAuth:
+    def __init__(
+        self,
+        api_base: ApiBase,
+        sync_client: SyncClient,
+        async_client: AsyncClient,
+    ) -> None:
+        self.api_base = api_base
+        self.sync_client = sync_client
+        self.async_client = async_client
+
+    def attach(
+        self,
+        provider: str,
+        user_id: Optional[str] = None,
+        session_token: Optional[str] = None,
+        session_jwt: Optional[str] = None,
+    ) -> AttachResponse:
+        """Generate an OAuth Attach Token to pre-associate an OAuth flow with an existing Stytch User. Pass the returned `oauth_attach_token` to the same provider's OAuth Start endpoint to treat this OAuth flow as a login for that user instead of a signup for a new user.
+
+        Exactly one of `user_id`, `session_token`, and `session_jwt` must be provided to identify the target Stytch User.
```

Comment:
docs nit - I'll open a PR for it later 
```suggestion
        Exactly one of `user_id`, `session_token`, or `session_jwt` must be provided to identify the target Stytch User.
```